### PR TITLE
Shorten the implementation of violin().

### PR DESCRIPTION
**Pull request by anntzer**
Just some standard rewriting.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -8100,15 +8100,17 @@ def violin(self, vpstats, positions=None, vert=True, widths=0.5,
...
         # Statistical quantities to be plotted on the violins
         means = []
         mins = []
         maxes = []
         medians = []
-        quantiles = np.asarray([])
+        quantiles = []
+
+        ppmins = []  # left end of quantiles lines
+        ppmaxs = []  # right end of quantiles lines
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py:8113]: Can we make better names? There's `ppmins` and `pmins`, but `ppmaxs` and `pmaxes`, and I don't even know what the `p` prefix means.

Comment (anntzer) [reply lib/matplotlib/axes/_axes.py:8113]: sure

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -8129,84 +8129,56 @@ def violin(self, vpstats, positions=None, vert=True, widths=0.5,
...
+            artists['cmeans'] = perp_lines(means, *line_ends)
+        if showextrema:  # Render extrema
+            artists['cmaxes'] = perp_lines(maxes, *line_ends)
+            artists['cmins'] = perp_lines(mins, *line_ends)
+            artists['cbars'] = par_lines(positions, mins, maxes)
+        if showmedians:  # Render medians
+            artists['cmedians'] = perp_lines(medians, *line_ends)
+        if quantiles:  # Render quantiles: each width is repeated qlen times.
+            idxs = [idx for idx, qlen in enumerate(qlens) for _ in range(qlen)]
+            artists['cquantiles'] = perp_lines(quantiles, *line_ends[:, idxs])
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py:8181]: Is this equivalent to?
```suggestion
            artists['cquantiles'] = perp_lines(quantiles, *np.repeat(line_ends, qlens, axis=1))
```

Comment (anntzer) [reply lib/matplotlib/axes/_axes.py:8181]: yes, that's better

**Review by QuLogic — APPROVED** [review]

**Review by efiring — APPROVED** [review]

**Pull request merged**